### ISSUE TITLE
fail run zftool after phar create

### ISSUE THREAD

**Issue by Userator** [zendframework/ZFTool#137]
```
userator@desktop:~ git clone https://github.com/zendframework/ZFTool.git ./zftool
...
userator@desktop:~ cd ./zftool
userator@desktop:~/zftool cd composer install
...
userator@desktop:~/zftool$ ./bin/create-phar
...
userator@desktop:~/zftool$ ./bin/zftool.phar 
PHP Fatal error:  Uncaught exception 'Zend\ModuleManager\Listener\Exception\RuntimeException' with message 'Could not find a valid ServiceManager for RoutePluginManager' in /home/kurshin/zftool/vendor/zendframework/zend-modulemanager/src/Listener/ServiceListener.php:202
Stack trace:
#0 /home/kurshin/zftool/vendor/zendframework/zend-eventmanager/src/EventManager.php(271): Zend\ModuleManager\Listener\ServiceListener->onLoadModulesPost(Object(Zend\ModuleManager\ModuleEvent))
#1 /home/kurshin/zftool/vendor/zendframework/zend-eventmanager/src/EventManager.php(143): Zend\EventManager\EventManager->triggerListeners(Object(Zend\ModuleManager\ModuleEvent))
#2 /home/kurshin/zftool/vendor/zendframework/zend-modulemanager/src/ModuleManager.php(129): Zend\EventManager\EventManager->triggerEvent(Object(Zend\ModuleManager\ModuleEvent))
#3 /home/kurshin/zftool/vendor/zendframework/zend-mvc/src/Application.php(264): Zend\ModuleManager\ModuleManager->loadModules()
#4 phar:///home/kurshin/zftool/bin/zftool.phar/zf.php(2): Zend\Mvc\Application::i in /home/kurshin/zftool/vendor/zendframework/zend-modulemanager/src/Listener/ServiceListener.php on line 202
```

**Comment by bierik** [comment]
Same problem here using mac os 10.12.1:

```
PHP Fatal error:  Uncaught Zend\ModuleManager\Listener\Exception\RuntimeException: Could not find a valid ServiceManager for RoutePluginManager in phar:///usr/local/bin/zftool.phar/vendor/zendframework/zend-modulemanager/src/Listener/ServiceListener.php:2
Stack trace:
#0 phar:///usr/local/bin/zftool.phar/vendor/zendframework/zend-eventmanager/src/EventManager.php(2): Zend\ModuleManager\Listener\ServiceListener->onLoadModulesPost(Object(Zend\ModuleManager\ModuleEvent))
#1 phar:///usr/local/bin/zftool.phar/vendor/zendframework/zend-eventmanager/src/EventManager.php(2): Zend\EventManager\EventManager->triggerListeners(Object(Zend\ModuleManager\ModuleEvent))
#2 phar:///usr/local/bin/zftool.phar/vendor/zendframework/zend-modulemanager/src/ModuleManager.php(2): Zend\EventManager\EventManager->triggerEvent(Object(Zend\ModuleManager\ModuleEvent))
#3 phar:///usr/local/bin/zftool.phar/vendor/zendframework/zend-mvc/src/Application.php(2): Zend\ModuleManager\ModuleManager->loadModules()
#4 phar:///usr/local/bin/zftool.phar/zf.php(2): Zend in phar:///usr/local/bin/zftool.phar/vendor/zendframework/zend-modulemanager/src/Listener/ServiceListener.php on line 2

Fatal error: Uncaught Zend\ModuleManager\Listener\Exception\RuntimeException: Could not find a valid ServiceManager for RoutePluginManager in phar:///usr/local/bin/zftool.phar/vendor/zendframework/zend-modulemanager/src/Listener/ServiceListener.php:2
Stack trace:
#0 phar:///usr/local/bin/zftool.phar/vendor/zendframework/zend-eventmanager/src/EventManager.php(2): Zend\ModuleManager\Listener\ServiceListener->onLoadModulesPost(Object(Zend\ModuleManager\ModuleEvent))
#1 phar:///usr/local/bin/zftool.phar/vendor/zendframework/zend-eventmanager/src/EventManager.php(2): Zend\EventManager\EventManager->triggerListeners(Object(Zend\ModuleManager\ModuleEvent))
#2 phar:///usr/local/bin/zftool.phar/vendor/zendframework/zend-modulemanager/src/ModuleManager.php(2): Zend\EventManager\EventManager->triggerEvent(Object(Zend\ModuleManager\ModuleEvent))
#3 phar:///usr/local/bin/zftool.phar/vendor/zendframework/zend-mvc/src/Application.php(2): Zend\ModuleManager\ModuleManager->loadModules()
#4 phar:///usr/local/bin/zftool.phar/zf.php(2): Zend in phar:///usr/local/bin/zftool.phar/vendor/zendframework/zend-modulemanager/src/Listener/ServiceListener.php on line 2
```

**Comment by BasBastian** [comment]
Old issue I solved recently. It is caused by choosing too liberal way of composer's dependency versioning. There were too many `>=` showing lack of experience on dependency management subject. If anythig would break, it will break with MAJOR. Selecting `>= 2.2.2` as lower number boundary (that implicates there is no upper limit for version) causes some serious issues if anyone chnages their API upgrading MAJOR to let others know, that there are some major changes ( some of them might be not backward-compatible ). 

TL; DR - just remove `>=` signs from ZFTools composer.json and perform `composer update`.

In your app consider as well some changes might break your app.

**Comment by BasBastian** [comment]
My bad. That actually wasn't the whole truth. It seems composer.json is lacking entry for zendframework or one of it's dependencies. 

Short and pretty bad fix is to add zendframework as depedency. But then zftool.phar becomes 8MB file and generating it takes a while.

We have to inspect it and search for real problem.
